### PR TITLE
avoid integer value retention event version overflow.

### PR DESCRIPTION
**Pull request by Roiocam**
References #31717

The result type is Long, upcast Integer to Long should fix it.

### CODE REVIEW

**Comment by lightbend-cla-validator** [comment]
Hi @Roiocam,

Thank you for your contribution! We really value the time you've taken to put this together.

We see that you have signed the Lightbend Contributors License Agreement before, however, the CLA has changed since you last signed it.
Please review the new CLA and sign it before we proceed with reviewing this pull request:

[https://www.lightbend.com/contribute/cla](https://www.lightbend.com/contribute/cla)

File: akka-persistence-typed/src/main/scala/akka/persistence/typed/internal/RetentionCriteriaImpl.scala [hunk 1]
```diff
@@ -26,14 +26,14 @@ import akka.persistence.typed.scaladsl
 
   def deleteUpperSequenceNr(lastSequenceNr: Long): Long = {
     // Delete old events, retain the latest
-    math.max(0, lastSequenceNr - (keepNSnapshots * snapshotEveryNEvents))
+    math.max(0, lastSequenceNr - (keepNSnapshots.asInstanceOf[Number].longValue() * snapshotEveryNEvents))
```
Comment (johanandren) [inline akka-persistence-typed/src/main/scala/akka/persistence/typed/internal/RetentionCriteriaImpl.scala:29]:
```suggestion
    math.max(0, lastSequenceNr - (keepNSnapshots.toLong * snapshotEveryNEvents))
```
And for the other one as well

Comment (Roiocam) [reply akka-persistence-typed/src/main/scala/akka/persistence/typed/internal/RetentionCriteriaImpl.scala]: Thanks!

**Review by johanandren — APPROVED** [review]
LGTM, thanks!

**Review by patriknw — APPROVED** [review]
LGTM, thanks

**Pull request merged**